docs/user/openstack/affinity.md

```diff
@@ -0,0 +1,136 @@
+# How to set affinity rules for workers at install-time
+
+The Installer automatically creates Control plane nodes with a
+"soft-anti-affinity" policy. This means that each server will be created on a
+separate host, if enough hosts are available.
+
+The Installer does not set any affinity policy on Compute nodes; the choice is
+left to the user.
+
+We are going to create an OpenStack `server group` with an `anti-affinity`
+policy. Then we are going to modify the `MachineSet` definitions during the
+installation process, so that when `openshift-install create cluster` is run,
+the Compute nodes will directly be created with the given policy.
+
+### Step 1: create the Server group
+
+In OpenStack Compute (Nova), scheduler policies are set using Server groups.
+
+The first step is then to create the Server group.
+
+There are four possible policies:
+* affinity
+* soft-affinity
+* anti-affinity
+* soft-anti-affinity
```

Comment:
I think it depends on how nova is configured, see https://docs.openstack.org/nova/latest/admin/configuration/schedulers.html and the `enabled_filters` flag in nova.conf.

I checked in my OSP environment and I have these filters enabled by default:
`enabled_filters=AvailabilityZoneFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter`
I know that operators can change that during their deployments of OpenStack and it makes me wonder if it can have an impact on what policies can be used here when creating a server group.
If that would be the case, maybe we can just add a note here saying that it's up to the user to check if whether or not the filter is enabled in Nova.

cc @mdbooth

Reply:
Nevermind, actually only the ServerGroupAffinityFilter needs to be enabled (default), but still an operator could deploy Nova and disable ServerGroupAffinityFilter (for some reason), and therefore it wouldn't be possible to create server groups with these policies, AFIU.

Reply:
Changed the wording to reflect that not all policies might be available in all clusters.